demo/app/main-view-model.ts

```diff
@@ -48,27 +48,38 @@ export class HelloWorldModel extends Observable {
     };
 
     setTimeout(() => {
-      console.log('setupCallListener!');
+      console.log('Registering call listeners');
       setupCallListener(callListener);
-    }, 10000)
+    }, 15000)
 
-    // listener for push notifications (incoming calls)
-    const pushListener = {
-      onPushRegistered(accessToken, deviceToken) {
-        dialogs.alert('push registration succeded');
-      },
-      onPushRegisterFailure (error) {
-        dialogs.alert(`push registration failed: ${error}`);
-      }
-    };
+  
 
-    setupPushListener(pushListener);
+    setTimeout(() => {
+        // listener for push notifications (incoming calls)
+      const pushListener = {
+        onPushRegistered(accessToken, deviceToken) {
+          dialogs.alert('push registration succeded');
+        },
+        onPushRegisterFailure (error) {
+          dialogs.alert(`push registration failed: ${error}`);
+        },
+
+        onIncomingCall(customParameters) {
+          return {
+            from: customParameters.from_name
+          }
+        }
+      };
+      console.log('Registering push listeners');
+      setupPushListener(pushListener);
+    },20000);
```

Comment:
I think there are some linter errors among your changes. For example `},20000);` to `}, 20000)`;.

I suggest you sync changes from master, as I [fixed some linters](https://github.com/msaelices/nativescript-twilio/commit/9d452a6660e0029d09c4df52b77f55888b14ddd7) errors there, and after that, execute `npm run tslint` and fix the tslint errors.

Reply:
BTW, why is the `setTimeout` needed? could you add a comment explaining that?